File: app/src/main/java/com/commit451/gitlab/navigation/RoutingNavigator.java
package com.commit451.gitlab.navigation;

import android.net.Uri;

/**
 * Interface called when routing in the Routing activity
 */
public interface RoutingNavigator {
    void onRouteToIssue(String projectNamespace, String projectName, String issueIid);
    void onRouteToCommit(String projectNamespace, String projectName, String commitSha);
    void onRouteToMergeRequest(String projectNamespace, String projectName, String mergeRequestId);
    void onRouteToProject(String namespace, String projectId);
    void onRouteToBuild(String projectNamespace, String projectName, String buildNumber);
    void onRouteToMilestone(String projectNamespace, String projectName, String milestoneNumber);
    void onRouteUnknown(Uri uri);
}

File: app/src/main/java/com/commit451/gitlab/navigation/RoutingNavigator.kt
package com.commit451.gitlab.navigation

import android.net.Uri

/**
 * Interface called when routing in the Routing activity
 */
interface RoutingNavigator {
    fun onRouteToIssue(projectNamespace: String, projectName: String, issueIid: String)
    fun onRouteToCommit(projectNamespace: String, projectName: String, commitSha: String)
    fun onRouteToMergeRequest(projectNamespace: String, projectName: String, mergeRequestIid: String)
    fun onRouteToProject(projectNamespace: String, projectName: String)
    fun onRouteToBuild(projectNamespace: String, projectName: String, buildNumber: String)
    fun onRouteToMilestone(projectNamespace: String, projectName: String, milestoneNumber: String)
    fun onRouteUnknown(uri: Uri?)
}
